sensorserviceutils/sensorserviceutils.go

```diff
@@ -383,16 +719,25 @@ func (p *parser) fullTimestamp(month, day, partialTimestamp string) (int64, erro
 	// If the bug report event log begins near the end of a year, and rolls over
 	// to the next year, the event would have taken place in the year preceding
 	// the reference year.
-	if int(p.referenceMonth)-parsedMonth < -1 {
+	if p.referenceMonth-month < -1 {
```

Comment:
can month be both larger or smaller than referenceMonth?

Reply:
Suppose we have a bugreport that records logs of the device during 2019.12.30 - 2020.01.01, the bug report is taken on 2020.01.01, so the reference month is January. but the registration history section may also include information from 2019.12.30-2020.01.01. This is the case where the reference month is smaller than the parsed month. Under such a condition, we need to decrement the "year" by 1 to create the correct timestamp for the log history.

On the other hand, if the device history records events during 2020.01.31 - 2020.02.01, the reference month is February. This is the case where the reference month is larger than the parsed month, and no changes to the "year" variable will be needed.

Most of the time we are expecting the reference month to be the same as the parsed month.